dashboard/lib/projects_list.rb

```diff
@@ -42,5 +84,35 @@ def get_project_row_data(student, project, channel_id)
         updatedAt: project_value['updatedAt'],
       }.with_indifferent_access
     end
+
+    def fetch_published_projects_all_types(limit)
+      PUBLISHED_PROJECT_TYPES.map do |type|
+        [type, fetch_published_projects_by_type(type, limit)]
+      end.to_h
+    end
+
+    def fetch_published_projects_by_type(project_type, limit, offset = 0)
+      PEGASUS_DB[:storage_apps].
+        where(state: 'active', project_type: project_type).
+        exclude(published_at: nil).
+        order(Sequel.desc(:published_at)).
+        limit(limit).
+        offset(offset).
+        map(&method(:get_published_project_data))
+    end
+
+    def get_published_project_data(project)
+      project_value = project[:value] ? JSON.parse(project[:value]) : {}
+      channel_id = storage_encrypt_channel_id(project[:storage_id], project[:id])
+      {
+        channel: channel_id,
+        name: project_value['name'],
+        thumbnailUrl: project_value['thumbnailUrl'],
+        # Note that we are using the new :project_type field rather than extracting
+        # it from :value. :project_type might not be present in unpublished projects.
+        type: project[:project_type],
+        publishedAt: project[:published_at]
+      }.with_indifferent_access
+    end
```

Comment:
These field names are chosen to match the field names returned by `get_project_row_data` above, since both data formats are likely to be used to populate the same `ProjectsList.jsx` and `ProjectCard.jsx` components in the UI.